src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -130,7 +131,10 @@ def convert_project_definition_to_v2(
         # since the file won't be re-read as it would be for a permanent conversion
         definition_v2 = render_definition_template(data, {}).project_definition
     else:
-        definition_v2 = ProjectDefinitionV2(**data)
+        # Context is needed, as the Pydantic model here will be evaluated only once
+        # and without context, the mixins won't be applied
+        with context({}):
+            definition_v2 = ProjectDefinitionV2(**data)
```

Comment:
Should we explicitly use here 
```suggestion
            definition_v2 = render_definition_template(data, {}).project_definition
```